src/components/organize/campaigns/CamapignCard.tsx

```diff
@@ -3,58 +3,44 @@ import { useRouter } from 'next/router';
 import { Card, CardActions, CardContent, Link, Typography } from '@material-ui/core';
 import { FormattedDate, FormattedMessage as Msg } from 'react-intl';
 
-import { getNaiveDate } from 'utils/dateUtils';
+import { removeOffset } from 'utils/dateUtils';
 import { ZetkinCampaign, ZetkinEvent } from 'types/zetkin';
 
 interface CampaignCardProps {
     campaign: ZetkinCampaign;
-    events: ZetkinEvent[];
-    upcomingEvents: ZetkinEvent[];
+    firstEvent?: ZetkinEvent;
+    lastEvent?: ZetkinEvent;
+    numOfUpcomingEvents?: number;
```

Comment:
@djbusstop I'm curious to hear your thoughts on this interface.

I think it would be nicer (cleaner, less risk for duplication of code etc) if the `CampaignCard` component accepted only two props: `campaign` and `events` (only the events belonging to the campaign).

The component could then contain the logic to find the first/last events, which ones are upcoming etc.

That would make the component much easier to use (consider the amount of code that would be moved out of the `campaigns/index.tsx` page) and also more clearly (IMO) separate the concerns of the containing page and the card component.

Do you think this makes sense?